File: DuDoong-Infrastructure/src/main/java/band/gosrock/infrastructure/config/ses/AwsSesConfig.java
package band.gosrock.infrastructure.config.ses;

import com.amazonaws.auth.AWSCredentials;
import com.amazonaws.auth.AWSStaticCredentialsProvider;
import com.amazonaws.auth.BasicAWSCredentials;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
import software.amazon.awssdk.auth.credentials.AwsCredentials;
import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.ses.SesAsyncClient;

@Configuration
public class AwsSesConfig {

    @Value("${aws.access-key}")
    private String accessKey;

    @Value("${aws.secret-key}")
    private String secretKey;

    @Bean
    public SesAsyncClient sesAsyncClient() {
        AwsBasicCredentials awsBasicCredentials = AwsBasicCredentials.create(this.accessKey,
            this.secretKey);
        return SesAsyncClient.builder()
            .credentialsProvider(StaticCredentialsProvider.create(awsBasicCredentials))
            .region(Region.US_WEST_2)
            .build();
    }
}

File: DuDoong-Infrastructure/src/main/kotlin/band/gosrock/infrastructure/config/ses/AwsSesConfig.kt
package band.gosrock.infrastructure.config.ses

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider
import software.amazon.awssdk.regions.Region
import software.amazon.awssdk.services.ses.SesClient

@Configuration
class AwsSesConfig(
    @Value("\${aws.access-key}") private val accessKey: String,
    @Value("\${aws.secret-key}") private val secretKey: String,
) {
    @Bean
    fun sesClient(): SesClient {
        val awsBasicCredentials = AwsBasicCredentials.create(accessKey, secretKey)
        return SesClient.builder()
            .credentialsProvider(StaticCredentialsProvider.create(awsBasicCredentials))
            .region(Region.US_WEST_2)
            .build()
    }
}
